woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -129,13 +129,15 @@ def test_init_accessor_with_schema(sample_df):
     assert head_df.ww.schema is None
     head_df.ww.init(schema=schema)
 
+    assert head_df.ww.schema is schema
```

Comment:
I think in this case we have head_df and schema_df both sharing the same underlying schema, so that changes to either one will be propagated to the other. Is this the behavior we want or should we also break that linkage here?

Reply:
Currently this is intentional. I was thinking originally that it should be up to the user to create a new object since I'm not sure that it's expected that Woodwork would create a new object at init. 

The situation where users are likely to pass in a schema themselves is when the Schema gets invalidated with an operation. In that case, users will want to take the original Schema and change it around until it's valid. Since schemas are mutable, they'll need to copy the schema or they'll be changing it for the original DataFrame as well--this creates a new object anyway. So I'm not sure when users would be passing a Schema from another DataFrame through without having to first make a new object since they should be going through the accessor for those.

But whether or not Woodwork creates a new object, we need to give users a way to copy a Schema if Schemas aren't going to be immutable.

Reply:
Reading the initial issue, I was under the impression that we would create a new schema object during init as well. I don't have a strong preference here, but we might want to make sure the behavior is clear in the documentation.

Reply:
I guess it occurred to me when I saw the Series Accessor getattr PR that it would make sense to create the new Table schema at the getattr instead of having it in the init. (if we do decide to have a new object at accessor init then we’ll need a corresponding change in the series accessor to copy metadata and semantic tags inits)

Will definitely make sure to mention in the `init` that the schema passed in gets used and that a new object is not created so users should be careful about passing in a schema that's used elsewhere.

Reply:
Just thinking out loud here a bit, but what if we were to update the user facing `.schema` property on the table accessor to return a new schema object with copies of the dictionary. That way, any time the user is trying to get the schema to modify it, like in the scenario that you described above, they are working with a copy. That way, users don't have to worry about these side effects and we don't make a new object during init unnecessarily.

Reply:
I like that!

Reply:
Added the copying behavior for the schema property and updated relevant tests to make sure we were referencing `_schema` if necessary